core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -150,44 +150,52 @@ object EntityDecoder {
     * that recoverable errors are returned as a
     * [[DecodeResult.failure]], or that system errors are raised in `F`.
     */
-  def decodeBy[F[_]: Applicative, T](r1: MediaRange, rs: MediaRange*)(
-      f: Media[F] => DecodeResult[F, T]): EntityDecoder[F, T] = new EntityDecoder[F, T] {
-    override def decode(m: Media[F], strict: Boolean): DecodeResult[F, T] =
+  // def decodeBy[F[_]: Applicative, T](r1: MediaRange, rs: MediaRange*)(
+  //     f: M[F] => DecodeResult[F, T]): EntityDecoder[F, T] = new
+      
+  abstract class DecodeByMediaRange[F[_]: Applicative, T](r1: MediaRange, rs: MediaRange*) extends EntityDecoder[F, T] {
```

Comment:
This seems like the other big breaking change.  Anybody who wrote their own decoders with `decodeBy` is losing.  I can't think of a way out of this.